packages/dds/tree/src/tree/treeTextFormat.ts

```diff
@@ -170,11 +171,18 @@ export function genericTreeKeys<T>(tree: GenericFieldsNode<T>): readonly FieldKe
         }
         return (Object.getOwnPropertyNames(global) as GlobalFieldKey[]).map(symbolFromKey);
     }
+
+    // If this node contains an empty array, it should return no keys.
+    const localKeys =
+        local[EmptyKey] !== undefined && local[EmptyKey].length === 0
```

Comment:
there is no reason `EmptyKey` should get ant special handling here. Its just one of many possible local field keys, and in has no special semantics at this level. Some schema  and schema aware APIs might give special meaning to specific names, but this code should not do so.

This file should not need any changes as part of this PR.

Reply:
I elected to change this function rather than the implementations of firstNode/nextNode because it seemed easier. Let me know if this is wrong.